include/mbedtls/pkcs5.h

```diff
@@ -59,7 +59,7 @@ extern "C" {
  * \param pwdlen     length of password
  * \param data       data to process
  * \param datalen    length of data
- * \param output     output buffer
+ * \param output     output buffer. must be of size \p datalen
```

Comment:
this is only true for decryption tho

Reply:
It's good to document this (it's really a bug that it wasn't documented), but we need the documentation to be correct! `datalen` is not correct.

Under the hood, `mbedtls_pkcs5_pbes2` calls `mbedtls_cipher_crypt` with the same buffer lengths, so the same constraint applies. From the documentation of `mbedtls_cipher_crypt` (I just changed the parameter names):

> This must be able to hold at least `datalen + block_size`. This must not be the same buffer as `input`.

This documentation is in fact overly pessimistic and I suspect that many users don't actually respect this constraint. In practice, the output buffer only needs to be large enough for the actual output. That's still larger than the input for encryption.

Reply:
Merged work has now addressed this lack of documentation (https://github.com/Mbed-TLS/mbedtls/commit/708d78f80b8b3c3cda9f46977a67625149c30591)